src/libraries/System.Runtime.InteropServices/tests/Ancillary.Interop/DefaultCaching.cs

```diff
@@ -2,14 +2,17 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 // Implementations of the COM strategy interfaces defined in Com.cs that we would want to ship (can be internal only if we don't want to allow users to provide their own implementations in v1).
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 
 namespace System.Runtime.InteropServices.Marshalling
 {
     internal sealed unsafe class DefaultCaching : IIUnknownCacheStrategy
     {
-        // [TODO] Implement some smart/thread-safe caching
-        private readonly Dictionary<RuntimeTypeHandle, IIUnknownCacheStrategy.TableInfo> _cache = new();
+        // We limit the concurrency level to 1 writer as we believe this is the most common scenario and limiting the concurrency level
+        // significantly reduces memory overhead.
+        // We set the default capacity a similar CoreCLR's COM-interop's RCW cache size.
```

Comment:
Can you point me to where we have a capacity of 16 in the build in? I can't find that value anywhere.

Reply:
We have a capacity of 8 in the inline cache in the built-in system, but I wanted to give a larger capacity here to ensure we have a decent load factor in the hash table.

Reference links for the built-in system:

https://github.com/dotnet/runtime/blob/e13f0dc1e0327b5d0cd0602b55ee257ef554f0a1/src/coreclr/vm/runtimecallablewrapper.h#L532

https://github.com/dotnet/runtime/blob/e13f0dc1e0327b5d0cd0602b55ee257ef554f0a1/src/coreclr/vm/comcache.h#L240-L255

https://github.com/dotnet/runtime/blob/e13f0dc1e0327b5d0cd0602b55ee257ef554f0a1/src/coreclr/vm/runtimecallablewrapper.h#L79

Reply:
Ah. Those aren't for the same cache. Those entries are for the inner lookup table attached to each RCW to avoid QI look ups for _that_ RCW. The equivalent cache for what we are talking about here is [`RCWCache`](https://github.com/dotnet/runtime/blob/e13f0dc1e0327b5d0cd0602b55ee257ef554f0a1/src/coreclr/vm/runtimecallablewrapper.h#L1074-L1084). I'm fine using 16 if you want to start there but I would remove the mention of basing it off of the internal cache. Instead you could state - this is a reasonable first guess. Or perhaps instrument the runtime and run a common WinForms app or two and get a starting point from there. Up to you.